docs/proposals/auth-gateway.md

```diff
@@ -0,0 +1,57 @@
+---
+title: "Authentication Gateway"
+linkTitle: "Authentication Gateway"
+weight: 1
+slug: "auth-gateway"
+---
+
+- Author: [Doğukan Teber](https://github.com/dogukanteber)
+- Date: March 2023
+- Status: Proposed
+
+## Overview
+
+If you run Cortex for multiple tenants you need to identify your tenants every time they send metrics or query them. This is needed to ensure that metrics can be ingested and queried separately from each other. For this purpose, the Cortex microservices require you to pass a Header called `X-Scope-OrgID`. Unfortunately, the Prometheus Remote write API has no config option to send headers and for Grafana you must provide a data source to do so. Therefore the Cortex k8s manifests suggest deploying an NGINX server inside of each tenant which acts as a reverse proxy. Its sole purpose is proxying the traffic and setting the `X-Scope-OrgID` header for your tenant.
+
+## Proposal
+
+We propose to solve this problem by adding a gateway that can be considered the entry point for all requests towards Cortex. This gateway will be responsible for handling multi-tenancy features without complicated proxy configuration.
+
+
+Here is an example configuration:
+
+```yaml
+
+basic:
```

Comment:
We will need more auth, but basic auth is a good start.

Reply:
My pleasure. It is decided the develop auth gateway in a different repository. The repo is already created. You can see it [here](https://github.com/cortexproject/auth-gateway).

I want to support different authentication types but I will use a Bearer token in the first release.